app/workers/kafka_listener_worker.rb

```diff
@@ -0,0 +1,29 @@
+require 'manageiq-messaging'
+
+class KafkaListenerWorker
+  include Sidekiq::Worker
+
+  def initialize(topic)
+    @topic = topic
+    @portal_ref = 'generic_1'
+  end
+
+  def start
+    Thread.new { KafkaListenerWorker.perform_async(@topic, @ref) }
```

Comment:
@bzwei Sidekiq has a worker framework and this listener worker is going to listen to Kafka Topics and push that data into Redis so Sidekiq Workers can run. This has to be run in a new thread because

https://github.com/ManageIQ/manageiq-messaging/blob/99bdc3844901dbe8f524fe8d84651bca8452e749/lib/manageiq/messaging/kafka/topic.rb#L18

Is a blocking call and it prevents us from creating multiple listeners. We have 2 workers one for approval and one for topology listening for 2 different topics and based on the messages coming in they call the correct SideKiq workers to do the job